### DISCUSSION TITLE
Create facade from an existing dashbaord

### DISCUSSION THREAD

**Discussion by v41e** [cdklabs/cdk-monitoring-constructs#469] [Q&A] [answered]
Is there a way to create a new monitoring facade but using an existing dashboard ?

I have different project and i would like to keep widget / alarms locally on each project but would like to have all widgets in a single dashboard

**Comment by ghost** [comment, marked as answer]
Unfortunately not. However, you could use the dashboard source to help build out the corresponding CDK code with the same metrics/widgets.

**Reply by v41e** [reply]
ok thanks, do you think i can open a feature request or there is no way it can be implemented and it's probably simpler to just put all widgets in a single stack with a single facade?

**Reply by ghost** [reply]
I think I misunderstood the question and thought you meant converting a non-CDK-managed dashboard to be managed via CDK using a facade (similar to the [EC2 Console-to-Code preview](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/console-to-code.html)).

Now assuming you mean you already have a facade creating dashboards and you want another facade that contributes to the same dashboard from the other facade: this also isn't really available out of the box, although you could try:

1. Creating the facade but disabling dashboards
2. Grab references to all of the created segments' `widgets()` and then manually adding it to the other facade's dashboard

Not particularly clean, although I'm not sure what the motivation for having separate facade instances is either.

**Reply by v41e** [reply]
I have several APIs (API Gateway), databases (Aurora, DynamoDB) in different git repositories. I would like to have in each repository a `monitor-stack` whose purpose is to produce alarms, and ideally widgets in an (external) dashboard.

But I prefer to have it all in one dashboard and would like to keep the monitoring code in each stack.

Or maybe this is not the idiomatic way to do this and it is better in my case to have another repository dedicated to monitoring with dynamic settings related to APIs and databases?

What are the recommendations in this scenario?

**Reply by ghost** [reply]
This issue seems related: https://github.com/aws/aws-cdk/issues/15334

I don't think there's a great/recommended way to effectively maintain a single dashboard with resources across stacks/CDK applications without actually importing everything into a single stack and doing it there (especially since the library depends on references to L2 constructs for the most part), although I do wish there was a better way.
